packages/penrose-core/src/compiler/Style.ts

```diff
@@ -1718,13 +1749,21 @@ const insertLabels = (trans: Translation, labels: LabelMap): Translation => {
     name: string,
     fieldDict: FieldDict
   ): [string, FieldDict] => {
-    let label: string = labels[name];
-    if (!label) {
-      label = name;
-      // Note that there's no reason for a shape to have no label; by default, it will be its substance name
+    let labelRes = labels.get(name);
+
+    if (!labelRes) {
+      // We skip here, to avoid putting spurious labels in for namespaces in the translation.
+      return [name, fieldDict];
+    }
+
+    let label;
+    if (labelRes.isJust()) {
+      label = labelRes.value;
+    } else {
+      // TODO: Distinguish between no label and empty label?
```

Comment:
Is this related to #474?

Reply:
Nah, just wondering if something doesn't get a label in Substance, should it have no label at all, or if its label should be the empty string. So far the latter seems safer for Style.